### PR TITLE
MULE-9546 JMSMessage-to-object-transformer doesn't propagate DataType

### PR DESCRIPTION
**Pull request by gsfernandes**

### CODE REVIEW

File: transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import javax.jms.JMSException;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+
+import org.junit.Test;
+
+public class JmsObjectToMessageDataTypeTransformerTestCase extends AbstractJmsFunctionalTestCase
+{
+    public static final int TIMEOUT = 3000;
```
Comment (pablokraan) [inline transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java]: This constant is not used

File: transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    protected String getConfigFile()
+    {
+        return "integration/jms-object-to-message-transformer-data-type-test-case.xml";
+    }
+
+    @Test
+    public void testDataTypeTransformation() throws Exception
+    {
+        send(sendTextMessageScenario);
+        FlowAssert.verify("message-to-string-flow");
```
Comment (pablokraan) [inline transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java]: Add static import

File: transports/jms/src/test/resources/integration/activemq/connector-jms-object-to-message-transformer-data-type-test-case.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (pablokraan) [inline transports/jms/src/test/resources/integration/activemq/connector-jms-object-to-message-transformer-data-type-test-case.xml]: We don't include license headers in the XML configs

**Comment by gsfernandes** [comment]
@pablokraan, PR is now ready for review.

File: transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Test
+    public void testDataTypeTransformation() throws Exception
+    {
+        send(sendTextMessageScenario);
+        verify("message-to-string-flow");
+    }
+
+    Scenario sendTextMessageScenario = new NonTransactedScenario()
```
Comment (pablokraan) [inline transports/jms/src/test/java/org/mule/transport/jms/integration/JmsObjectToMessageDataTypeTransformerTestCase.java]: Make this field local on testDataTypeTransformation

File: transports/jms/src/test/resources/integration/activemq/connector-jms-object-to-message-transformer-data-type-test-case.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  ~ Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
```
Comment (pablokraan) [inline transports/jms/src/test/resources/integration/activemq/connector-jms-object-to-message-transformer-data-type-test-case.xml]: Remove license from XML files

**Pull request merged**